### PR TITLE
Initiate Conformance Test Suite

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes

This commit initiates the OSS conformance test suite. It sets the
standards for incrementing the V1 conformance test with the two simple
test cases.

/kind misc
part of #7828 
# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [n/a] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [n/a] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by JeromeJu** [comment]
It might also be helpful for us to create the `/kind conformance` given the amount of work in the future we might have for this - each major/minor version could have its own patch etc.

**Comment by JeromeJu** [comment]
cc @tektoncd/core-maintainers PTAL 🙏

**Comment by JeromeJu** [comment]
cc @afrittoli  @chitrangpatel @vdemeester @Yongxuanzhang  thanks for your time in advance 🙏

File: test/conformance/util.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	for _, cond := range prStatus.Conditions {
+		if cond.Type == "Succeeded" {
+			if string(cond.Status) != expectedSucceededStatus {
+				return fmt.Errorf("Expect vendor service to populate Condition %s but got: %s", expectedSucceededStatus, cond.Status)
+			}
+			if cond.Reason != expectedReason {
+				return fmt.Errorf("Expect vendor service to populate Condition Reason %s but got: %s", expectedReason, cond.Reason)
+			}
+
+			hasSucceededConditionType = true
```
Comment (Yongxuanzhang) [inline test/conformance/util.go:66]: can we just return nil here?

**Review by Yongxuanzhang — APPROVED** [review]

**Review by chitrangpatel — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]

**Pull request merged**